MSVC build failure: Windows ERROR macro conflicts with MessageType::ERROR

## Description

On Windows with MSVC (Visual Studio 17 2022), compilation fails with **C2589** and **C2062** errors in source files that reference `MessageType::ERROR` in switch statements or expressions.

### Root cause

`<windows.h>` defines `ERROR` as a preprocessor macro (value `0`). When MSVC compiles code like `case MessageType::ERROR:`, the preprocessor expands it to `case MessageType::0:`, which is invalid C++.

The enum definition in `types.h` is already protected with `#undef ERROR` before the `MessageType` enum. However, `.cpp` files that include additional system/platform headers may transitively re-include `<windows.h>` (or `<wingdi.h>`) **after** `types.h`, which re-defines the `ERROR` macro. By the time the compiler reaches switch statements using `MessageType::ERROR`, the macro is active again.

### Affected files

- `src/someip/message.cpp` (line 400 — `case MessageType::ERROR:`)
- `src/someip/types.cpp` (lines 33, 78, 95 — `MessageType::ERROR`)
- `src/rpc/rpc_server.cpp` (line 180 — `MessageType::ERROR`)

### Build error

```
message.cpp(400,27): error C2589: 'constant': illegal token on right side of '::'
message.cpp(400,14): error C2062: type '<error type>' unexpected
```

### Proposed fix

Add `#ifdef ERROR` / `#undef ERROR` / `#endif` after all `#include` directives in each affected `.cpp` file. This ensures the macro is neutralized regardless of transitive Windows header inclusions.

### Environment

- **Compiler**: MSVC 17.14 (Visual Studio 17 2022)
- **OS**: Windows (GitHub Actions runner `windows-latest`)
- **Python binding build**: opensomeip-python using CMake submodule

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -166,6 +166,16 @@ target_link_libraries(someip-tp PRIVATE someip-core PRIVATE someip-transport)
 # Create convenience aliases for backward compatibility
 add_library(someip-common ALIAS someip-core)
 
+# On Windows, <windows.h> pulls in <wingdi.h> which defines ERROR as a macro.
+# This collides with MessageType::ERROR.  NOGDI suppresses that header entirely;
+# WIN32_LEAN_AND_MEAN trims other unused Win32 sub-headers.
+if(WIN32)
+    foreach(_tgt someip-core-obj someip-e2e-obj someip-serialization
+                 someip-transport someip-rpc someip-sd someip-events someip-tp)
+        target_compile_definitions(${_tgt} PRIVATE NOGDI WIN32_LEAN_AND_MEAN)
+    endforeach()
+endif()
+
 # Set library properties (only on real targets, not aliases)
 set_target_properties(someip-core someip-serialization someip-transport PROPERTIES
     VERSION ${PROJECT_VERSION}

